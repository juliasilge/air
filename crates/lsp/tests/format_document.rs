mod fixtures;

use fixtures::test_client::init_test_client;
use fixtures::tower_lsp_test_client::TestClientExt;
use lsp::documents::Document;

#[tests_macros::lsp_test]
async fn test_format() {
    let mut client = init_test_client().await;

    #[rustfmt::skip]
    let doc = Document::doodle(
"
1
2+2
3 + 3 +
3",
    );

    let formatted = client.format_document(&doc).await;
    insta::assert_snapshot!(formatted);

    client
}
