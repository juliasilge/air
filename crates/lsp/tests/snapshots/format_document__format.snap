---
source: crates/lsp/tests/format_document.rs
expression: formatted
---
1
2 + 2
3 + 3 + 3
